src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/MigrationClientService.kt

```diff
@@ -0,0 +1,55 @@
+package uk.gov.justice.digital.hmpps.authorizationserver.service
+
+import org.springframework.core.convert.ConversionService
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.authorizationserver.data.model.AuthorizationConsent
+import uk.gov.justice.digital.hmpps.authorizationserver.data.model.Client
+import uk.gov.justice.digital.hmpps.authorizationserver.data.model.ClientConfig
+import uk.gov.justice.digital.hmpps.authorizationserver.data.repository.AuthorizationConsentRepository
+import uk.gov.justice.digital.hmpps.authorizationserver.data.repository.ClientConfigRepository
+import uk.gov.justice.digital.hmpps.authorizationserver.data.repository.ClientRepository
+import uk.gov.justice.digital.hmpps.authorizationserver.resource.MigrationClientRequest
+import java.time.LocalDate
+
+@Service
+class MigrationClientService(
+  private val clientRepository: ClientRepository,
+  private val clientConfigRepository: ClientConfigRepository,
+  private val authorizationConsentRepository: AuthorizationConsentRepository,
+  private val clientIdService: ClientIdService,
+  private val conversionService: ConversionService,
+  private val clientsService: ClientsService,
+) {
+
+  @Transactional
+  fun addClient(migrationClientRequest: MigrationClientRequest) {
+    val clientList = clientIdService.findByBaseClientId(migrationClientRequest.clientId!!)
+    if (clientList.isNotEmpty()) {
+      throw ClientAlreadyExistsException(migrationClientRequest.clientId)
+    }
+    var client = conversionService.convert(migrationClientRequest, Client::class.java)
+
+    client = client?.let { clientRepository.save(it) }
```

Comment:
Do we need the ? operator here? The client returned from the conversion service will never be null?

Reply:
agreed , removed now